metadata/params/params.go

```diff
@@ -39,6 +40,30 @@ func astRoot(component, snippet string) (ast.Node, error) {
 	return parser.Parse(tokens)
 }
 
+// sanitizeComponent puts quotes around a component name if it contains special
+// characters.
+func sanitizeComponent(component string) string {
+	if utils.HasSpecialCharacter(component) {
+		return fmt.Sprintf(`"%s"`, component)
+	}
+	return component
+}
+
+func hasComponent(component string, field ast.ObjectField) bool {
+	// case foo: {...}
+	if field.Id != nil && string(*field.Id) == component {
+		return true
+	}
+	// case "foo-bar": {...}
+	if field.Expr1 != nil {
+		c, isStringLiteral := field.Expr1.(*ast.LiteralString)
+		if isStringLiteral && c.Value == component {
+			return true
+		}
+	}
```

Comment:
The `ast.ObjectField` type has a member, `Kind`, which you can switch on here to determine whether `Id` is populated or not. You should need to handle only the `ObjectFieldID` and `ObjectFieldStr` cases:

```
switch field.Kind {
case ObjectFieldStr: [... expr1 logic here ...]
case ObjectFieldID: [... id logic here ...]
}
```

Reply:
Also this will break if the literal string is a block. We should probably switch and emit the correct string type just in case.